server/shared/src/main/scala/org/http4s/server/middleware/Timeout.scala

```diff
@@ -31,6 +31,15 @@ object Timeout {
     * duration if the service has not yet responded.  If the timeout
     * fires, the service's response is canceled.
     *
+    * Note, if effects of service responding is uncancelable
+    * (e.g. if service uses `MonadCancel#uncancelable` under the hood) and has exceeded
+    * the timeout, then an expected behavior is:
+    * <ul>
+    *   <li> effects will be completed naturally (regardless of how long it takes), </li>
+    *   <li> after that, the timeout response will be returned. </li>
+    * </ul>
+    * To get more insights on effect canceling, dig into the Cats-Effect documentation.
```

Comment:
It would be nice if we had a more specific link for this, but I suspect it doesn't exist! 😬

Reply:
```suggestion
    * To get more insights on effect cancellation, dig into the Cats-Effect documentation.
```